content/docs/user-guide/project-structure/pipelines-files.md

```diff
@@ -0,0 +1,420 @@
+    metrics:
+      - accuracy.txt:
+          cache: false
+    plots:
+      - auc.json:
+          cache: false
+```
+
+> `cache: false` is typical here, since they're small enough for Git to version
+> directly.
+
+The commands in `dvc metrics` and `dvc plots` help you display and compare
+metrics and plots.
+
+## Templating
+
+⚠️ This feature is only available in DVC 2.0 ⚠️
+
+`dvc.yaml` supports a templating format to insert values from different sources
+in the YAML structure itself. These sources can be
+[parameters files](/doc/command-reference/params), or `vars` defined in
+`dvc.yaml` instead.
+
+> Note that this parameterization feature is only supported via manual edition
+> of `dvc.yaml` and incompatible with `dvc run`.
+
+Let's say we have `params.yaml` (default params file) with the following
+contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+Those values can be used anywhere in `dvc.yaml` with the `${}` _substitution
+expression_:
+
+<!-- prettier-ignore-start -->
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --thresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
+```
+<!-- prettier-ignore-end -->
+
+DVC will track simple param values (numbers, strings, etc.) used in `${}` (they
+will be listed by `dvc params diff`).
+
+Alternatively, values for substitution can be listed as top-level `vars` like
+this:
+
+```yaml
+vars:
+  - models:
+      us:
+        threshold: 10
+  - desc: 'Reusable description'
+
+stages:
+  build-us:
+    desc: ${desc}
+    cmd: python train.py --thresh ${models.us.threshold}
+```
+
+> DVC merges values from params files and `vars` when possible. For example,
+> `{"grp": {"a": 1}}` merges with `{"grp": {"b": 2}}`, but not with
+> `{"grp": {"a": 7}}`.
+
+> Note that values from `vars` are not tracked like parameters.
+
+To load additional params files, list them in the top `vars`, in the desired
+order, e.g.:
+
+> Their paths will be evaluated based on
+> [`wdir`](/doc/user-guide/dvc-files/dvc.yaml#accepted-fields), if one given.
+
+```yaml
+vars:
+  - params.json
+  - myvar: 'value'
+  - config/myapp.yaml
+```
+
+(ℹ️) Note that the default `params.yaml` file is always loaded first, if
+present.
+
+It's also possible to specify what to include from additional params files, with
+a `:` colon:
+
+```yaml
+vars:
+  - params.json:clean,feats
+
+stages:
+  featurize:
+    cmd: ${feats.exec}
+    deps:
+      - ${clean.filename}
+    outs:
+      - ${feats.dirname}
+```
+
+Stage-specific values are also supported, with inner `vars`. You may also load
+additional params files locally. For example:
+
+```yaml
+stages:
+  build-us:
+    vars:
+      - params.json:build
+      - model:
+          filename: 'model-us.hdf5'
+    cmd: python train.py ${build.epochs} --out ${model.filename}
+    outs:
+      - ${model.filename}
+```
+
+⚠️ Known limitations of local `vars`:
```

Comment:
I don't think this warning is useful as both of them are expected.

Reply:
Hmmm. Why would users expect or assume any of this? The one about `wdir` is a logical consequence of the loading order but that's a complex thing to grasp (for example it wasn't obvious to me at first and I had to ask). The one about `foreach` has some internal reason (I don't even remember it now) pretty much imposible for a user to know.

Plus we have all sorts of usage warnings throughout docs. I see this in line with that.